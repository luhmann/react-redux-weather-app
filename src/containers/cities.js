import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { loadCities } from '../actions'
import City from '../components/city'

class Cities extends Component {
  componentDidMount() {
    this.props.loadCities(['Berlin', 'Hamburg', 'MÃ¼nchen', 'New York', 'Tokio'])
  }

  render() {
    return (
      <div className="cities">
        {Object.keys(this.props.cities).map(cityName => {
          const city = this.props.cities[cityName]

          return (
            <City
              key={city.name}
              name={city.name}
              degrees={city.degrees}
              iconUrl={city.iconUrl}
              isLoading={city.isLoading}
            />
          )
        })}
      </div>
    )
  }
}

const mapStateToProps = ({ weather }) => ({
  cities: weather.cities,
})

const mapDispatchToProps = {
  loadCities,
}

export { Cities }

export default connect(mapStateToProps, mapDispatchToProps)(Cities)
