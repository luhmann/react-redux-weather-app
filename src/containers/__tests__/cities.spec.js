import React from 'react'
import { shallow, mount } from 'enzyme'

import { Cities } from '../cities'

describe('Container: Cities', () => {
  const mockCities = {
    hamburg: {
      degrees: -5,
      name: 'Hamburg',
      iconUrl: 'http://example.org/icons/hamburg.png',
      isLoading: false,
    },
    berlin: {
      degrees: 10,
      name: 'Berlin',
      iconUrl: 'http://example.org/icons/berlin.png',
      isLoading: false,
    },
    maui: {
      degrees: 28,
      name: 'Maui',
      iconUrl: 'http://example.org/icons/maui.png',
      isLoading: false,
    },
    paris: {
      degrees: 18,
      name: 'Paris',
      iconUrl: 'http://example.org/icons/paris.png',
      isLoading: false,
    },
  }

  it('should render cities passed from redux store', () => {
    const loadCitiesDispatch = jest.fn()

    const subject = shallow(
      <Cities cities={mockCities} loadCities={loadCitiesDispatch} />
    )

    expect(subject).toMatchSnapshot()
  })

  it('should dispatch the loadCities action-creator', () => {
    const loadCitiesDispatch = jest.fn()

    const subject = mount(
      <Cities cities={mockCities} loadCities={loadCitiesDispatch} />
    )

    expect(loadCitiesDispatch.mock.calls.length).toBe(1)
    expect(loadCitiesDispatch.mock.calls[0][0]).toEqual([
      'Berlin',
      'Hamburg',
      'MÃ¼nchen',
      'New York',
      'Tokio',
    ])
  })
})
