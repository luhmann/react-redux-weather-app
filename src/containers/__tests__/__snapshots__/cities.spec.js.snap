// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: Cities should render cities passed from redux store 1`] = `
<div
  className="cities"
>
  <City
    degrees={-5}
    iconUrl="http://example.org/icons/hamburg.png"
    isLoading={false}
    name="Hamburg"
  />
  <City
    degrees={10}
    iconUrl="http://example.org/icons/berlin.png"
    isLoading={false}
    name="Berlin"
  />
  <City
    degrees={28}
    iconUrl="http://example.org/icons/maui.png"
    isLoading={false}
    name="Maui"
  />
  <City
    degrees={18}
    iconUrl="http://example.org/icons/paris.png"
    isLoading={false}
    name="Paris"
  />
</div>
`;
